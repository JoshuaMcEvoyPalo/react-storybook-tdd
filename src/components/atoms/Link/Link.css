.link {
    font-family: 'Ubuntu', serif;
    font-style: normal;
    font-weight: 300;
    font-size: 16px;
    line-height: 150%;
    text-decoration-line: underline;
    color: #112D4E;
}