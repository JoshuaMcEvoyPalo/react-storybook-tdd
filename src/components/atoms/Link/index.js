import React from 'react'
import './Link.css'

const Link = ({ link, text }) => {
  return (
    <a data-testid="link" className="link" href={link}>{text}</a>
  )
}

export default Link
