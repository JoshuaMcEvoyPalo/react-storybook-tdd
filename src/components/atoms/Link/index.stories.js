import Link from '.'

export default {
  title: 'Atoms / Link',
  component: Link,
  argTypes: { onClick: { action: 'clicked' } },
}

const Template = (args) => <Link {...args} />

export const Default = Template.bind({})
Default.args = {
  text: 'Forgotten your username?',
  link: '/another-page',
}