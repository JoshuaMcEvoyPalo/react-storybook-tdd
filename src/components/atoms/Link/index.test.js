import renderer from 'react-test-renderer'
import Link from '.'

describe('Link tests', () => {
  it('Should render Link correctly', () => {
    const tree = renderer.create(<Link text="Forgotten your username?" link="/another-page" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})