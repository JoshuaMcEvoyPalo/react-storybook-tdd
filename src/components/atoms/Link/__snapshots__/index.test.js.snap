// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link tests Should render Link correctly 1`] = `
<a
  className="link"
  data-testid="link"
  href="/another-page"
>
  Forgotten your username?
</a>
`;
